src/file/tokio_compat_file.rs

```diff
@@ -516,6 +527,8 @@ impl AsyncWrite for TokioCompatFile {
             return Poll::Ready(Err(sftp_to_io_error(cancel_error())));
         }
 
+        this.write_len.store(0, Ordering::Relaxed);
```

Comment:
After changing `write_len` from `AtomicUsize` => `usize`:

```suggestion
        *this.write_len = 0;
```